shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java

```diff
@@ -52,17 +96,419 @@ public static void addMetaTag(
+            return;
+        }
+        Organization organization = new Organization();
+        organization.setIdentifier(List.of(identifier));
+        Reference orgReference = createOrganizationReference(bundle, organization);
+        patientIdentifier.setAssigner(orgReference);
+    }
+
+    public static String getPID3_4Value(Bundle bundle) {
+        Identifier identifier = getPID3_4Identifier(bundle);
+        if (identifier == null) {
+            return null;
+        }
+        return identifier.getValue();
+    }
+
+    public static void setPID3_4Value(Bundle bundle, String value) {
+        Identifier identifier = getPID3_4Identifier(bundle);
+        if (identifier == null) {
+            return;
+        }
+        identifier.setValue(value);
+    }
+
+    // PID-3.5 - Identifier Type Code
+    public static Coding getPID3_5Coding(Bundle bundle) {
+        Identifier identifier = getPID3Identifier(bundle);
+        if (identifier == null) {
+            return null;
+        }
+        return identifier.getType().getCodingFirstRep();
+    }
+
+    public static void setPID3_5Coding(Bundle bundle, Coding coding) {
+        Identifier identifier = getPID3Identifier(bundle);
+        if (identifier == null) {
+            return;
+        }
+        identifier.setType(new CodeableConcept().addCoding(coding));
+    }
+
+    public static String getPID3_5Value(Bundle bundle) {
+        Coding coding = getPID3_5Coding(bundle);
+        if (coding == null) {
+            return null;
+        }
+        return coding.getCode();
+    }
+
+    public static void setPID3_5Value(Bundle bundle, String value) {
+        Coding coding = getPID3_5Coding(bundle);
+        if (coding == null) {
+            return;
+        }
+        coding.setCode(value);
+    }
+
+    // PID-5 - Patient Name
+    public static Extension getPID5Extension(Bundle bundle) {
+        Patient patient = getPIDPatient(bundle);
+        if (patient == null) {
+            return null;
+        }
+        HumanName name = patient.getNameFirstRep();
+        return name.getExtensionByUrl(HapiHelper.EXTENSION_XPN_HUMAN_NAME_URL);
+    }
+
+    public static void setPID5Extension(Bundle bundle) {
+        Patient patient = getPIDPatient(bundle);
+        if (patient == null) {
+            return;
+        }
+        HumanName name = patient.getNameFirstRep();
+        name.addExtension(new Extension(HapiHelper.EXTENSION_XPN_HUMAN_NAME_URL));
+    }
+
+    // PID-5.7 - Name Type Code
+    public static String getPID5_7Value(Bundle bundle) {
+        Extension extension = getPID5Extension(bundle);
+        if (extension == null || !extension.hasExtension(HapiHelper.EXTENSION_XPN7_URL)) {
+            return null;
+        }
+        return extension
+                .getExtensionByUrl(HapiHelper.EXTENSION_XPN7_URL)
+                .getValue()
+                .primitiveValue();
+    }
+
+    public static void setPID5_7Value(Bundle bundle, String value) {
```

Comment:
This is only called by test code.  Let's remove it.  If you want this as a helper method to make testing easier, add it to (or create new) a helper class that is just inside the `test/groovy/` area.

Reply:
Moved to test package